packages/components/src/components/domainproperties/list/ListDesignerPanels.tsx

```diff
@@ -42,8 +42,17 @@ export class ListDesignerPanels extends React.PureComponent<Props, State> {
     constructor(props) {
         super(props);
 
+        let model = props.initModel;
+        if (props.initModel.name) {
+            const fields = props.initModel.domain.fields;
+            const pkField = fields.findIndex(i => (i.isPrimaryKey));
+            const pkFieldLocked = fields.get(pkField).merge({ lockType:"PKLocked", required:true }) as DomainField;
```

Comment:
The constructor here seems like an odd place to modify the lockType for a domain field. Any chance this could be moved into the ListModel.create()? That would seem like a better place to me for this mapping of locktype and required bit to happen. The actions.ts fetchListDesign() function does end up in ListModel.create(), so I think that should work.